File: domain/src/main/kotlin/com/sidequest/domain/model/ActionPlan.kt
package com.sidequest.domain.model

import kotlinx.serialization.Serializable

/**
 * An ordered breakdown of an [ActionItem] into smaller steps.
 */
@Serializable
data class ActionPlan(
    val id: String,
    val actionItemId: String,
    val subActions: List<SubAction>,
    val sync: SyncMeta,
)

/**
 * A single step within an [ActionPlan]. [order] defines its position in the
 * sequence; [completed] tracks whether the step is done.
 */
@Serializable
data class SubAction(
    val id: String,
    val text: String,
    val order: Int,
    val completed: Boolean,
)

File: ios/SideQuestKit/Sources/SideQuestKit/Models/ActionPlan.swift
import Foundation

/// An ordered breakdown of an `ActionItem` into smaller steps
/// (contract: `backend/api/openapi.yaml` → `ActionPlan`).
///
/// `subActions` is an ordered list of 1...100 steps (Req 9.8). Progress and the
/// "mark complete" prompt (Req 9.9, 9.10) are derived by the portable domain
/// logic in a later task.
public struct ActionPlan: Codable, Identifiable, Equatable {

    public var id: String
    public var actionItemId: String
    public var subActions: [SubAction]
    public var sync: SyncMeta

    public init(
        id: String,
        actionItemId: String,
        subActions: [SubAction],
        sync: SyncMeta
    ) {
        self.id = id
        self.actionItemId = actionItemId
        self.subActions = subActions
        self.sync = sync
    }
}

/// A single step within an `ActionPlan`
/// (contract: `backend/api/openapi.yaml` → `SubAction`).
///
/// `order` defines its position in the (contiguous) sequence; `completed`
/// tracks whether the step is done.
public struct SubAction: Codable, Identifiable, Equatable {

    public var id: String
    public var text: String
    public var order: Int
    public var completed: Bool

    public init(id: String, text: String, order: Int, completed: Bool) {
        self.id = id
        self.text = text
        self.order = order
        self.completed = completed
    }
}
